src/miniscript/satisfy.rs

```diff
@@ -105,11 +106,19 @@ pub trait Satisfier<Pk: MiniscriptKey + ToPublicKey> {
     }
 
     /// Assert whether an relative locktime is satisfied
+    ///
+    /// NOTE: If a descriptor mixes time-based and height-based timelocks, the implementation of
+    /// this method MUST only allow timelocks of either unit, but not both. Allowing both could cause
+    /// miniscript to construct an invalid witness.
```

Comment:
In d29c2984e60e41e3872ac23b8b59e8b25fea88a2:

nit: I would suggest
```
NOTE: this method must ONLY allow time-based or height-based relative timelocks, except for the value
zero which is valid as both a height or a time. Allowing both could cause miniscript to consruct an invalid
witness for descriptors which mix height- and time-based timelocks. (We do not consider such descriptors
to be "sane" but they can still be processed by this library, e.g. by using `parse_insane` to parse them.)
```

And same below (but change "relative" to "absolute").

But I'm not certain about the zero comment. Can somebody more familiar with timelocks check on that?

Reply:
My main concern with the existing comment is that it doesn't make it clear that absolute and relative timelocks are considered separately; so you can have a height-based relative timelock and a time-based absolute timelock, for example.

Reply:
Also, these comments are currently on `Satisfier::check_older` and `Satisfier::check_after`. Should they also be on the equivalent methods in `AssetProvider`?